Fix remaining akka-testkit warnings

raboof opened the pull request:
Mostly in tests and 2.13

raboof committed:
Fix remaining akka-testkit warnings

Mostly in tests and 2.13

chbatey commented on akka-testkit/src/main/scala/akka/testkit/TestKit.scala:
```diff
@@ -919,6 +919,7 @@ trait TestKitBase {
  *
  * @since 1.1
  */
+@silent // 'early initializers' are deprecated on 2.13 and will be replaced with trait parameters on 2.14
```
issue for fixing it?

chbatey reviewed (approved):
LGTM

raboof committed:
Add issue link

johanandren reviewed (approved):
LGTM

raboof committed:
MiMa exclude for multi-node-testkit

We don't promise bincompat there anyway, but perhaps good to keep mima there
so at least we don't break things accidentally
